### PR TITLE
Set correct message types when using proxy interfaces

### PR DESCRIPTION
**Pull request by timbussmann**
Connects to Particular/V6Launch#68

Sends and replies set the message type of the generated proxy class instead of the interface which resulted in issues during message metadata lookup and the message conventions in case they were based on the type name.

repro was raised by @ramonsmits https://github.com/ramonsmits/docs.particular.net/pull/1 and this will be working correctly with these changes (tested locally).

@Particular/nservicebus-maintainers @ramonsmits please review

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/Basic/When_using_a_greedy_convention.cs [hunk 1]
```diff
@@ -5,24 +5,21 @@
...
+            var context = await Scenario.Define<Context>(c => { c.Id = Guid.NewGuid(); })
                 .WithEndpoint<Endpoint>(b => b.When((session, c) => session.SendLocal(new MyMessage
                 {
                     Id = c.Id
                 })))
                 .Done(c => c.WasCalled)
-                .Repeat(r => r
-                    .For(Transports.Default)
-                )
-                .Should(c => Assert.True(c.WasCalled, "The message handler should be called"))
```
Comment (danielmarbach) [inline src/NServiceBus.AcceptanceTests/Basic/When_using_a_greedy_convention.cs]: Nice cleanup

File: src/NServiceBus.Core.Tests/Unicast/MessageOperationsTests.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        // ReSharper disable once MemberCanBePrivate.Global
+        public interface IMyMessage
+        {
+        }
+
+        class MyMessage
+        {
+        }
+
+        class FakePipelineCache : IPipelineCache
```
Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/Unicast/MessageOperationsTests.cs]: I think this is not the only pipeline cache we have. Would it make sense in a separate PR to toolify this?

**Comment by danielmarbach** [comment]
:+1: Shall I merge? Looks good

File: src/NServiceBus.Core.Tests/Unicast/MessageOperationsTests.cs (hunk 2 above)
Comment (timbussmann) [inline src/NServiceBus.Core.Tests/Unicast/MessageOperationsTests.cs]: yeah I think there might be other places using similar approaches.

**Comment by timbussmann** [comment]
please do :tada:
